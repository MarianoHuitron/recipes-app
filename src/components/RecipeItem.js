import React from 'react'
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';
import LazyImage from './LazyImage'

const RecipeItem = ({ recipe }) => {
    
    const history = useHistory();

    const { strMeal, idMeal, strMealThumb } = recipe;
    return (
        <div className="col mb-4">
            <div className="card h-100">
                <LazyImage 
                    src={ strMealThumb }
                    alt={ `image-${strMeal}` }
                    styleClass="card-img-top"
                />
                <div className="card-body">
                    <h5 className="card-title"> {strMeal} </h5>
                   
                </div>
                <div className="card-footer">
                <button 
                    className="btn btn-block btn-secondary"  
                    onClick={() => history.push(`/view/${idMeal}`)}    
                >See More</button>
                </div>
            </div>
        </div>
    )
}

RecipeItem.propTypes = {
    recipe: PropTypes.object.isRequired
}

export default RecipeItem
