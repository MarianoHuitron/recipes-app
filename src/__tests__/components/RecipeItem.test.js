import { shallow } from 'enzyme';
import React from 'react';
import RecipeItem from '../../components/RecipeItem';

const mockHistory = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        push: () => mockHistory()
    })
}));

const recipe = {
    idMeal: "52874",
    strMeal: "Beef and Mustard Pie",
    strMealThumb: "http://imae.jg"
};


describe('Testing <RecipeItem />', () => {

    beforeEach(() => {
        jest.clearAllMocks();
    })

    const wrapper = shallow(
        <RecipeItem recipe={recipe} />
    );

    test('should render RecipeItem', () => { 
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.card-title').text().trim() ).toBe(recipe.strMeal);
    });

    test('should call history.push', () => {
        wrapper.find('button').prop('onClick')();
        expect( mockHistory).toHaveBeenCalled();
    })
    
    
})
