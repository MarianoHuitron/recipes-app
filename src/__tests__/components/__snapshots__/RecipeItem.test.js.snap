// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <RecipeItem /> should render RecipeItem 1`] = `
<div
  className="col mb-4"
>
  <div
    className="card h-100"
  >
    <LazyImage
      alt="image-Beef and Mustard Pie"
      src="http://imae.jg"
      styleClass="card-img-top"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
         
        Beef and Mustard Pie
         
      </h5>
    </div>
    <div
      className="card-footer"
    >
      <button
        className="btn btn-block btn-secondary"
        onClick={[Function]}
      >
        See More
      </button>
    </div>
  </div>
</div>
`;
